src/pages/Register/RegisterForm/EmailTextField.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import { TextField } from '@mui/material'
+import { UseFormRegister, FieldErrors } from 'react-hook-form'
+
+import { RegisterInputType } from './types'
+
+
+// eslint-disable-next-line no-useless-escape
+const EMAIL_REGEX = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
```

Comment:
reuse validation, components from login form

Reply:
check new commit [refactor: Extract textfields for reusability]
no error were seen, output the same
commit msg has description about what i did, please review

Reply:
If looking good, then i shall re-locate component into component space, so that login and forgot password can use Email text field too. Its logic doesn't differ from each other